GitHubContents.ps1

```diff
@@ -508,11 +556,37 @@ filter Add-GitHubContentAdditionalProperties
 
         if (-not (Get-GitHubConfiguration -Name DisablePipelineSupport))
         {
-            $elements = Split-GitHubUri -Uri $item.url
+            if ($item.url)
+            {
+                $uri = $item.html_url
+            }
+            else
+            {
+                $uri = $item.content.html_url
+            }
+
+            $elements = Split-GitHubUri -Uri $uri
             $repositoryUrl = Join-GitHubUri @elements
+
             Add-Member -InputObject $item -Name 'RepositoryUrl' -Value $repositoryUrl -MemberType NoteProperty -Force
+
+            $hostName = $(Get-GitHubConfiguration -Name 'ApiHostName')
+
+            if ($uri -match "^https?://$hostName/?([^/]+)/?([^/]+)/blob/?([^/]+)/?([^#]*)?$")
```

Comment:
```suggestion
            $hostName = $(Get-GitHubConfiguration -Name 'ApiHostName')
            if ($uri -match "^https?://$hostName/([^/]+)/([^/]+)/(blob|tree)/([^/]+)/([^#]*)?$")
```

I simplified the RegEx a little bit, and made it work for both individual files as well as when a directory list is returned (we won't have a filename, but we'll still have a branch name).

Reply:
Actually, even better...specify non-capturing groups so that you can just always use `$Matches[1]` and `$Matches[2]`:

```powershell
            if ($uri -match "^https?://(?:www\.|api\.|)$hostName/(?:[^/]+)/(?:[^/]+)/(?:blob|tree)/([^/]+)/([^#]*)?$")
```